CONTRIBUTING.md

```diff
@@ -28,6 +28,66 @@ use GitHub pull requests for this purpose. Consult
 [GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
 information on using pull requests.
 
+## Pull Request Title Convention
+
+We enforce a PR title convention to quickly indicate type and scope of a PR.
+We also parse it programmatically for changelog generation.
+
+PR titles should
+* be a user facing description of the change
+* be [conventional](https://www.conventionalcommits.org/en/v1.0.0/)
+* append fixed issue(s) at the end (if any)
+
+Examples:
+* `"fix(ui): fixes empty page. Fixes #1234"`
+* `"feat(backend): configurable service account. Fixes #1234, fixes #1235"`
+* `"chore: refactor some files"`
+* `"test: fix CI failure. Part of #1234"` // use "part of" when a PR is an work item of some issue, but shouldn't close the issue when merged.
+
+A quick summary of the conventions we care about:
+### PR Title Structure
+```
+<type>[optional scope]: <description> [Fixes #<issue-number>]
```

Comment:
Currently, the variable substitutions are inconsistent. Consider changing this to something like:

`<type>(<scope>): <description> [Fixes #<issue-number>] `

Also consider adding a section like the following below the  code block:

Replace the following:

*  **`<type>`**: The PR type describes the reason for the change, such as `fix` to indicate that the PR fixes an open issue. More information about PR types is available in the next section. 
*  **`<scope>`**: (Optional.) The PR scope describes the part of Kubeflow Pipelines that this PR changes, such as `frontend` to indicate that the change affects the user interface.
*  **`<description>`**: A user friendly description of this change.
*  **`[Fixes #<issues-number>]`**: (Optional.) Specifies the issues fixed by this PR.

Reply:
I guess sth not clear is that, there's a syntax that uses `<>` to specify something required, and `[]` to specify something optional, because scope is optional including the parenthesis, we cannot put parenthesis there.

The replace the following section is really helpful. Gonna add it, thanks!

Reply:
Please take a look at new revision